rmi/servercode/TransImpl/HashVault.java

```diff
@@ -0,0 +1,81 @@
+package TransImpl;
+
+import ResInterface.*;
+import ResImpl.*;
+
+import java.util.*;
+import java.io.*;
+import java.rmi.RemoteException;
+
+public class HashVault
+{
+  // The HashVault Class:
+  // Hashmap of transaction hashtable shadows, as well as our master shadows
+  // Each Key is unique int, the hashKey referenced in TransactionManager
+  // Each Value is a temporary or master hashtable
+  private static HashMap<Integer, RMHashtable> vault;
+  private static int default_tables = 4;
+
+
+	public HashVault(ArrayList<RMHashtable> masterRecords) {
+    this.vault = new HashMap<Integer, RMHashtable>();
+
+    // Add default RMHashtables
+    for (int i = 0; i < default_tables; i++) {
+      this.vault.put(i, masterRecords.get(i));
+    }
+
+	}
+
+	public void store(int hashKey, RMHashtable shadow) {
```

Comment:
add/update a hashtable

Reply:
Note that we call this when we're storing temporary shadows, or updating our master shadows. It's a pretty generic function